### ISSUE TITLE
Go back and jump to the original location occasionally

### ISSUE THREAD

**Issue by yanzhang0219** [iam4x/pokemongo-webspoof#167]
When I flip the Pokestop and try to catch some pokemons while I am in the autopilot, my character occasionally jumps back to the original location where I am staying. How to solve this? Or I cannot do anything during the autopilot? Thanks a lot. 

Rocky

**Comment by yanzhang0219** [comment]
This happens especially when I open the gym during autopilot. Thx

**Comment by NicholasHord** [comment]
I experience the same issue, not even sure if its something that can be fixed on the app end?

**Comment by yanzhang0219** [comment]
@NicholasHord Yes. And I'm so afraid I'm gonna have a soft banned due to this jumping back and forth.

**Comment by yanzhang0219** [comment]
Update:

When this happens, I check the xcode and the whole 'Debug' menu gets gray. Then after I re-build and run it again, the 'Debug' menu turns available and I can choose the 'simulate location' -->'PokemonLocation' again. Then my character jumps to the location before.

**Comment by TeXnicians** [comment]
I suspect that might be caused by unstable connection of your USB cable. My USB port is old and this happens occasionally. But my friend's iPhone is new and turns out to be very stable. But this is just a speculation.

**Comment by yanzhang0219** [comment]
@TeXnicians Thanks a lot. I will try another cable. But this issue happens to me when I open the gym and catch some pokemons. That means if I leave the main map screen and enter the gym or catching screen, when I come back, it happens. So sad....  Could your friend battle in the gym without any issues?

**Comment by TeXnicians** [comment]
I think so. 
Technically, if your cable is not stable, when you touch the screen it causes tiny vibration and hence may affect the connection a little bit, right :)? Still just speculation.

**Comment by AbelIngrand** [comment]
It can come from the cable but it can also come from the background app quitting due to memory exhaustion (especially since it happens on gyms or pokestops). We're working on a real fix to keep the app launched in the background.

**Comment by yanzhang0219** [comment]
@AbelIngrand Thank you so much. It must be memory exhaustion. Hope you guys can figure it out.

**Comment by jhoye88** [comment]
Does the application stop(so you have to build and run it again in Xcode)?

**Comment by Takage** [comment]
Same problem here. Sometimes it is due to the App stopping sometimes it just jumps back without any reason and when I click on one of the arrows it jumps back to the location on the map

**Comment by mywebENUDD** [comment]
This is due to a background app issue that was not resolved properly in v1.1.1, A new commit has been merged into the project and hopefully will resolve the issue in v1.1.2

**Comment by Lumerias** [comment]
This is exactly what I talked about in #164 and I was told something about an mp3 in the background fix, or something. I have no idea. I'm not a programmer at all, sorry.

**Comment by mywebENUDD** [comment]
Yup, I think that was me who told you that, on my other account. But yes a code attempt to play audio in background was made but had a few bugs in it, someone submitted a PR with some fixes, we will see if it works the next time a .app is released.
It's alright if you're not a programmer, just understand that all github software in a way is in constant beta mode in one way or another and most likely contains bugs ;)

**Comment by Lumerias** [comment]
That, I totally understand! At this point, it's unplayable for me unless I just want to catch pokemon and collect stuff from pokestops. Every time I attempt to train or take over a gym, I hop back to my original location and get kicked out of the gym. There is an in-game glitch where if you are kicked out of a gym you cannot go back into the gym immediately after and retry to take it over, hopefully Niantic fixes that soon.

**Comment by Takage** [comment]
It is not only that I would argue this bug classifies as a Sev1 since it can ban us all due to location jumping

**Comment by mywebENUDD** [comment]
The use of this software is against Pokemon Go's TOS, using it at all can potentially ban you. But I do agree this bug is high priority, good thing someone added a potential fix recently, can't wait to use it after work.

**Comment by yanzhang0219** [comment]
Update:

I have tried a larger memory device, like iPad, and it works well. Maybe, iPhone6s/ 6Plus still works fine. You guys can try up.

**Comment by Takage** [comment]
Nope I am using a iPhone 6 Plus, killed all apps and still running in those problems

**Comment by herocertified** [comment]
Known issue. Supposed to be fixed in 1.1.2
See #121 and #178

**Comment by yanzhang0219** [comment]
@herocertified Got it. Thanks a lot. By the way, when is the expected date for 1.1.2?

**Comment by Takage** [comment]
Not fixed with 1.1.2. Still jumping around

**Comment by IPyandy** [comment]
not fixed in 1.2.0 either

**Comment by TeXnicians** [comment]
I think this problem could be hard to resolve. If Pokemon Go occupies too much memory (especially when you are fighting in gym), the background apps could be affected. So the solution of this issue depends on the optimization of Pokemon Go rather than this app. The best way, for now, is to use a large memory device, such as iPad. I've never encountered the issue when using iPad but frequently encountered it on iPhone.

**Comment by Mardaneus86** [comment]
My experience is that if you go to a specific location, then unplug your device, and then rerun the Xcode project, it occasionaly jumps back to the location you had where you unplugged the phone. So that prevents you from jumping way too far. Not a solution, but a possible workaround to prevent you from being softbanned.
